### ISSUE TITLE
on fetch, env vars are not updated

### ISSUE THREAD

**Issue by cjdcordeiro** [nuvla/api-server#369]
To reproduce:

* deploy an app with an env var, without defaults...assign that env var at deployment time
* update the module to hardcode a default env var
* shutdown and fetch the module version from the deployment page
* restart the deployment, and you'll the end var is still the old one, not the new one

┆Issue is synchronized with this [Trello card](https://trello.com/c/BCrCiYvv) by [Unito](https://www.unito.io)

**Comment by 0xbase12** [comment]
It's on purpose and there no right answer to this question.

The current implementation is done as follow:
Try to keep values already defined in the running application. (User defined them, or working defaults)
If an env key is no more defined in the module, delete it.
If a new env key  is present, take it with it's default value.

**Comment by cjdcordeiro** [comment]
I think that makes sense, only if I shutdown and then restart the deployment.

But if I "fetch", in between those two actions, I should get the latest commit, as is...at least that what I expected...otherwise it will be inconsistent.

I did exactly that...created a new commit where only the env var default changed...then I did the fetch, but it was for nothing cause my change was not "fetched".

Maybe double check with the team
